pkg/api/datasources.go

```diff
@@ -385,10 +386,22 @@ func (hs *HTTPServer) CheckDatasourceHealth(c *models.ReqContext) {
 		return
 	}
 
+	var jsonDetails map[string]interface{}
 	payload := map[string]interface{}{
 		"status":      resp.Status.String(),
 		"message":     resp.Message,
-		"jsonDetails": resp.JSONDetails,
+		"jsonDetails": jsonDetails,
+	}
+
+	// Unmarshal JSONDetails if its not empty.
+	if len(resp.JSONDetails) > 1 {
+		err = json.Unmarshal(resp.JSONDetails, &jsonDetails)
```

Comment:
Why not unmarshal directly to `payload["jsonDetails"]`? I could be wrong, but I think it should be preferable (clearer, no copying).
```suggestion
		err = json.Unmarshal(resp.JSONDetails, &payload["jsonDetails"])
```

Reply:
:) I tried that first but GO didn't like that. 

```
pkg/api/datasources.go:398:42: cannot take the address of payload["jsonDetails"]
```
I guess its because `payload["jsonDetails"]` is an `interface{}`

Reply:
@bergquist Ah yeah, it can be a bit tricky to foresee these error messages. Does the following work (i.e. using the map directly and not a pointer, seem to recall it does work)?

```
json.Unmarshal(resp.JSONDetails, payload["jsonDetails"])
```

Reply:
Never mind, I tested and it doesn't work.